### PR TITLE
[cmake] Adding options for INSTALL and TEST

### PR DESCRIPTION
**Pull request by hdf89shfdfs**
Not all consumers of GSL automatically want to have this install logic.

It's good practice to gate install logic behind an option.
For an example look at magic_enum:
https://github.com/Neargye/magic_enum/blob/master/CMakeLists.txt

If the client wants to install GSL they still can. But they should ask
for it by overriding GSL_INSTALL.

### CODE REVIEW

**Comment by hdf89shfdfs** [comment]
@JordanMaples do the changes look acceptable?

**Comment by JordanMaples** [comment]
At a quick glance everything looks good. I'll do a deeper dive in this on Monday. 

Does adding the project options change how cmake needs to be invoked in the CLI? If so, would you mind adding a blurb in the readme about invoking these options.

File: cmake/guidelineSupportLibrary.cmake [hunk 1]
```diff
@@ -75,3 +78,55 @@ function(gsl_add_native_visualizer_support)
...
+            COMPATIBILITY SameMajorVersion
+        )
+    else()
+        write_basic_package_version_file(
+            ${CMAKE_CURRENT_BINARY_DIR}/Microsoft.GSLConfigVersion.cmake
+            COMPATIBILITY SameMajorVersion
+            ARCH_INDEPENDENT
+        )
+    endif()
+endfunction()
```
Comment (JordanMaples) [inline cmake/guidelineSupportLibrary.cmake:132]:
```suggestion
endfunction()

```

**Comment by hdf89shfdfs** [comment]
> 
> 
> At a quick glance everything looks good. I'll do a deeper dive in this on Monday.
> 
> Does adding the project options change how cmake needs to be invoked in the CLI? If so, would you mind adding a blurb in the readme about invoking these options.

It shouldn't affect the CLI

File: CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +25,10 @@ if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
     set(GSL_STANDALONE_PROJECT ON)
 endif()
 
+### Project options
+option(GSL_INSTALL "Generate and install gsl target" ${GSL_STANDALONE_PROJECT})
```
Comment (JordanMaples) [inline CMakeLists.txt:29]:
```suggestion
option(GSL_INSTALL "Generate and install GSL target" ${GSL_STANDALONE_PROJECT})
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -25,6 +25,10 @@ if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
...
+option(GSL_TEST "Build and perform gsl tests" ${GSL_STANDALONE_PROJECT})
```
Comment (JordanMaples) [inline CMakeLists.txt:30]:
```suggestion
option(GSL_TEST "Build and perform GSL tests" ${GSL_STANDALONE_PROJECT})
```

**Review by JordanMaples — APPROVED** [review]

**Comment by JordanMaples** [comment]
Maintainer's call: Merging this. I verified the changes locally and everything looks good to me. Thanks!

**Pull request merged**